packages/sdk-artifact/src/ArtifactClient.test.ts

```diff
@@ -0,0 +1,82 @@
+import { expect } from 'earljs'
+import { ReadStream } from 'fs'
+import { readFile, rm, writeFile } from 'fs/promises'
+import { Readable } from 'stream'
+
+import { mock } from './__test/mock'
+import { ArtifactClient } from './ArtifactClient'
+import { ArtifactApi } from './networking/ArtifactApi'
+import { HttpClient } from './networking/HttpClient'
+
+describe(ArtifactClient.name, () => {
+  const fileFath = 'file.json'
```

Comment:
can we use something like https://www.npmjs.com/package/tmp to create files in tmp dir and not worry about (currently brittle) cleanup?

Reply:
The brittle thing about current setup is that you can forget to clean up. And I would be all for using `tmp` if it would offer context syntax like:
```python
with open() as f:
    doStuff()
```
But it doesn't and there is still a chance we forget to clean up.
I might argue that it's even worse, cause now the file is in our repo, and we will notice it with `git status`, but we won't if it lives in `/tmp`.

Where is my logic flawed?

Reply:
Yes, or simply when test fails we will endup with some random files in the working dir. Perfect solution would be to move this to tmp + use after/before each hooks.